### ISSUE TITLE
Device Profile "Assigned Firmware" not synced from CE Docker 3.4

### ISSUE THREAD

**Issue by joinke** [thingsboard/thingsboard-edge#18]
I'm trying to perform OTA update from a Device connected to Edge.
The OTA firmware is visible in Edge and can be downloaded from the Edge UI.

However :
- The device profile "assigned firmware" field has not synced from the Thingsboard CE. 
- Also , if I assign a Firmware directly to the device from the Thingsboard CE , it does not sync this field to Edge.
- So, this means that when using the firmware url download , i get a 404 error, because the device does not have an assigned fw.
-
Workaround so far to be able to use the OTA firmware URL is to manually assign the firmware in the Edge UI.
(url : http://192.168.xx.xx:8080/api/v1/TOKEN/firmware?title=Tank_Level&version=1.0.27&size=0&chunk=1)

Am I missing something obvious here ?

Edge

![image](https://user-images.githubusercontent.com/38005255/182827714-47cb0d87-54ae-4860-8861-3a0618851192.png)

CE

![image](https://user-images.githubusercontent.com/38005255/182827921-7a8494b7-76ab-4641-83c6-dcb2e951cac9.png)

Edge Log when modifying the Assigned firmware of device profile on the CE 

2022-08-04 20:44:49,479 [TB-Scheduling-1] INFO  o.t.s.c.t.s.DefaultTransportService - Transport Stats: openConnections [0]
**2022-08-04 20:45:26,024 [grpc-default-executor-67] INFO  o.t.s.s.c.r.p.BaseCloudProcessor - Adding ATTRIBUTES_REQUEST/RELATION_REQUEST fd1ecb60-133d-11ed-908a-d922e38596b1 DEVICE**
2022-08-04 20:45:28,129 [sql-log-1-thread-1] INFO  o.t.s.dao.sql.TbSqlBlockingQueue - Queue-0 [TS] queueSize [0] totalAdded [2] totalSaved [2] totalFailed [0]
202

CE log when modifying assigned firmware of device profile on  CE

2022-08-04 10:45:23,917 [rule-dispatcher-3-3] INFO  o.t.rule.engine.action.TbLogNode -
Incoming message:
{"id":{"entityType":"DEVICE_PROFILE","id":"7dafec70-0ca2-11ed-b7a7-5d88a11bd8fd"},"createdTime":1658812755127,"tenantId":{"entityType":"TENANT","id":"6d046130-f474-11ec-b746-1bfa669193d7"},"name":"Water Level","description":"","image":null,"type":"DEFAULT","transportType":"DEFAULT","provisionType":"CHECK_PRE_PROVISIONED_DEVICES","defaultRuleChainId":null,"defaultDashboardId":null,"defaultQueueName":null,"profileData":{"configuration":{"type":"DEFAULT"},"transportConfiguration":{"type":"DEFAULT"},"provisionConfiguration":{"type":"CHECK_PRE_PROVISIONED_DEVICES","provisionDeviceSecret":"ayrubtxsal0l09mg9jra"},"alarms":null},"provisionDeviceKey":"s6ymhzoxhan5l7k9kwwq","firmwareId":{"entityType":"OTA_PACKAGE","id":"70f1fb40-13d2-11ed-ad99-cdc5dc232715"},"softwareId":null,"externalId":null,"default":false}
Incoming metadata:
{"user

**Comment by volodymyr-babak** [comment]
@joinke thanks for creating this ticket. 

This is a bug in the 3.4 release. 
The fix for this is going to be in the 3.4.1 release. 

Please use a workaround at the moment and assign firmware to the device using Edge UI. 
Sorry for any inconviences. 

Pull request to this issue is going to be created today/tomorrow.

**Comment by joinke** [comment]
Many thanks for replying superfast and for proposing alternative/fix !

**Comment by volodymyr-babak** [comment]
Fixed in the 3.4.1 release https://thingsboard.io/docs/edge/releases/#v341-august-19-2022

**Comment by joinke** [comment]
Hello @volodymyr-babak ,

I've tested this fix . The device firmware on the TB edge is now updated when I update the device profile in the TBCE. I can see that in the fw_tag, fw_version .. attributes. So, that solves the problem. One observation though , on the TB edge , in the device profile details , I can't see the Firmware details :

(both TB and TB edge are 3.4.1) (TB CE docker and TBEdge Ubuntu)
CE:
![image](https://user-images.githubusercontent.com/38005255/185728993-31033b68-7e87-4ace-a8e8-4a6a60ae534d.png)


Edge :
![image](https://user-images.githubusercontent.com/38005255/185728976-7cb1d73d-2fec-455b-ac66-da9558b5bbe3.png)
